kroxylicious-additional-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/DekContext.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption;
+
+import java.nio.ByteBuffer;
+
+/**
+ * A DekContext encapsulates an encryptor
+ * @param <K> The type of the KEK id
+ */
+final class DekContext<K> {
+    private final AesGcmEncryptor encryptor;
+    private final ByteBuffer prefix;
+    private final K kekId;
+
+    DekContext(K kekId,
+               ByteBuffer prefix,
+               AesGcmEncryptor encryptor) {
+        this.kekId = kekId;
+        this.prefix = prefix;
```

Comment:
if partition data contains many records, encrypt will be call more than once.

should prefix be a read-only buffer?
```suggestion
        this.prefix = prefix.asReadOnlyBuffer();
```

Reply:
This issue was resolve in another way.